massgov/mayflower
# DP-10056: Allow the compilation of a local mayflower artifacts directory

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

This PR changes the way we're compiling Mayflower Patternlab in order to make the mayflower artifacts possible to compile separately from the Patternlab instance.  In particular:

* The gulp tasks relating to patternlab are now dependent on the `dist:build` task (it's run automatically).  The `dist` folder is now just copied into the patternlab directory.
* The dist folder is now configurable using the `.env` file.   Copy `patternlab/styleguide/.env.example` to `patternlab/styleguide/.env` and edit the `MAYFLOWER_DIST` variable in that file.  In combination with https://github.com/massgov/mass/pull/2661, this makes it possible to see your mayflower changes in Drupal immediately.
* The `dist` folder is now exactly what gets committed to `mayflower-artifacts` *and* what gets pushed to `npm`, meaning we no longer have to build the artifacts and npm package separately.
* All deployment mechanisms have been stripped out of gulp and moved to the circle config, making our Gulp scripting a lot easier to understand, and keeping all deployment actions in the same place (the Circle config).
* NPM publishing for both React and Patternlab has been converted so the `package.json` files no longer need a manual version bump each time.  This is done automatically in CircleCI based on the git tag.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-10056)
- [Github issue]()

## Steps to Test


1. Preview mayflower locally as normal.
2. Run `gulp --dist=/tmp/mayflower-artifacts` and verify that the folder is created, and that it has the same structure as our current [mayflower-artifacts](https://github.com/massgov/mayflower-artifacts) repo.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
